uptime/checker.go

```diff
@@ -0,0 +1,416 @@
+			val.lastChecked = (*upInfos)[i].checkedTime
+
+			// moving average calculation
+			newCount := val.avgLatency + 1
+			val.avgLatency = val.avgLatency * val.avgLatency / newCount + val.latency / newCount
+			val.avgLatency = newCount
+		}
+		healthInfos[addr] = val
+	}
+
+	return nil
+}
+
+func (u *UptimeChecker) multiAddrsUp(addrs *[]MultiAddr) []UpInfo {
+	upInfos := make([]UpInfo, 0)
+	for _, addr := range(*addrs) {
+		upInfos = append(upInfos, u.isUp(addr))
+	}
+	return upInfos
+}
+
+func (u *UptimeChecker) processReportedCheckers(ctx context.Context) error {
+	for {
+		if u.IsStop() {
+			break
+		}
+
+		state, err := Load(ctx, u.api, u.actorAddress, u.self)
+		if err != nil {
+			log.Errorw("cannot load state", "err", err)
+			continue
+		}
+
+		listToCheck, err := state.ListReportedCheckerNotVoted()
+		if err != nil {
+			log.Errorw("cannot list repored checkers not voted", "err", err)
+			continue
+		}
+
+		for toCheckPeerID, addrs := range(listToCheck) {
+			u.CheckChecker(ctx, toCheckPeerID, addrs)
+		}
+
+		time.Sleep(time.Duration(5) * time.Second)
+	}
+
+	return nil
+}
+
+func (u *UptimeChecker) monitorMemberNodes(ctx context.Context) error {
+	for {
+		if u.IsStop() {
+			break
+		}
+
+		state, err := Load(ctx, u.api, u.actorAddress, u.self)
+		if err != nil {
+			log.Errorw("cannot load state", "err", err)
+			continue
+		}
+
+		listToCheck, err := state.ListMembers()
+		if err != nil {
+			log.Errorw("cannot list members", "err", err)
+			continue
+		}
+
+		for _, toCheckActorID := range listToCheck {
+			addrs, err := state.ListMemberMultiAddrs(toCheckActorID)
+			if err != nil {
+				log.Errorw("cannot list member multi addrs", "actor", toCheckActorID, "err", err)
+				continue
+			}
+
+			log.Debugw("member info", "actor", toCheckActorID, "addrs", addrs)
+
+			u.CheckMember(toCheckActorID, addrs)
+		}
+
+		time.Sleep(time.Duration(5) * time.Second)
+	}
+
+	return nil
+}
+
+func (u *UptimeChecker) monitorCheckerNodes(ctx context.Context) error {
+	for {
+		if u.IsStop() {
+			break
+		}
+
+		state, err := Load(ctx, u.api, u.actorAddress, u.self)
+		if err != nil {
+			log.Errorw("cannot load state", "err", err)
+			continue
+		}
+
+		listToCheck, err := state.ListCheckers()
+		if err != nil {
+			log.Errorw("cannot list members", "err", err)
+			continue
+		}
+
+		for _, toCheckPeerID := range listToCheck {
+			addrs, err := state.ListCheckerMultiAddrs(toCheckPeerID)
+			if err != nil {
+				log.Errorw("cannot list member multi addrs", "peer", toCheckPeerID, "err", err)
+				continue
+			}
+
+			log.Debugw("toCheckPeerID addrs", "toCheckPeerID", toCheckPeerID, "addrs", addrs)
+
+			err = u.CheckChecker(ctx, toCheckPeerID, addrs)
+			if err != nil {
+				log.Errorw("cannot check checker", "peer", toCheckPeerID, "err", err)
+			}
+		}
+
+		time.Sleep(time.Duration(5) * time.Second)
+	}
+
+	return nil
+}
+
+func (u *UptimeChecker) execSync(ctx context.Context, method uint32, from address.Address, params []byte) error {
```

Comment:
What about calling this message `onChainMsg` and renaming the one below also? I think it makes the code a bit more readable.